kvm/src/lib.rs

```diff
@@ -53,15 +53,14 @@ pub struct Kvm {
 
 impl Kvm {
     /// Opens `/dev/kvm/` and returns a `Kvm` object on success.
-    pub fn new() -> Result<Self> {
-        // Safe because we give a constant nul-terminated string and verify the result.
-        let ret = unsafe { open("/dev/kvm\0".as_ptr() as *const c_char, O_RDWR) };
-        if ret < 0 {
-            return Err(io::Error::last_os_error());
-        }
-
+    /// # Arguments
+    ///
+    /// * `close_on_exec`: Open `/dev/kvm` using the `O_CLOEXEC` flag.
+    ///
+    pub fn new(close_on_exec: bool) -> Result<Self> {
```

Comment:
I agree it is not the best way to specify the flags. What I am trying to avoid here is to let the user set whatever flags for opening /dev/kvm because then we will also have to check if the flags are valid. Moreover, I would like to keep the same level of abstraction across modules/crates (I shall explain this in real life as I have difficulties in finding words right now). I will come with a better solution and let you know.

Reply:
I made the kvm constructor open /dev/kvm with close on exec by default. Another change is that the open function is now named open_with_cloexec and has a bool parameter. That should make it a bit more clear what the boolean is there for. Let me know if that's okay.